src/StringTools/StringTools.cs

```diff
@@ -0,0 +1,233 @@
+        /// </summary>
+        /// <param name="callback">The callback to remove.</param>
+        public static void UnregisterStringInterningCallback(TryInternStringDelegate callback)
+        {
+            lock (s_locker)
+            {
+                s_internStringCallbacks = s_internStringCallbacks.Where(existingCallback => existingCallback != callback).ToArray();
+            }
+        }
+
+        /// <summary>
+        /// Enables diagnostics in the interner. Call <see cref="CreateDiagnosticReport"/> to retrieve the diagnostic data.
+        /// </summary>
+        public static void EnableDiagnostics()
+        {
+            OpportunisticIntern.Instance.EnableStatisticsGathering();
+        }
+
+        /// <summary>
+        /// Retrieves the diagnostic data describing the current state of the interner. Make sure to call <see cref="EnableDiagnostics"/> beforehand.
+        /// </summary>
+        public static string CreateDiagnosticReport()
+        {
+            StringBuilder callbackReport = new StringBuilder();
+            callbackReport.AppendFormat("{0} with {1} string interning callbacks registered", nameof(Strings), s_internStringCallbacks.Length);
+            callbackReport.AppendLine();
+
+            return callbackReport.ToString() + OpportunisticIntern.Instance.FormatStatistics();
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Returns a <see cref="SpanBasedStringBuilder"/> instance back to the pool if possible.
+        /// </summary>
+        /// <param name="stringBuilder">The instance to return.</param>
+        internal static void ReturnSpanBasedStringBuilder(SpanBasedStringBuilder stringBuilder)
+        {
+            if (stringBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(stringBuilder));
+            }
+#if !NET35
+            s_stringBuilderPool.Return(stringBuilder);
+#endif
+        }
+
+        /// <summary>
+        /// Calls interning callbacks in sequence until one return true.
+        /// </summary>
+        /// <param name="candidate">The candidate internable string to invoke the callbacks with.</param>
+        /// <param name="interned">The resulting string.</param>
+        /// <returns></returns>
+        internal static bool CallStringInterningCallbacks(ref InternableString candidate, out string interned)
+        {
+            // We can read s_internStringCallbacks lock-less because it's declared volatile and RegisterStringInterningCallback
+            // and UnregisterStringInterningCallback write the field after initializing its content.
+            TryInternStringDelegate[] callbacks = s_internStringCallbacks;
+            foreach (TryInternStringDelegate callback in callbacks)
+            {
+                if (callback(ref candidate, out interned))
+                {
+                    return true;
+                }
+            }
```

Comment:
```suggestion
            return callbacks.Any(callback => TryInternStringDelegate(callback, out interned));
```
? Might change the value of interned, so you'd have to account for that when using it. (You already do, for now.)

Reply:
This code is gone, no more hardcoded strings.